packages/next/server/web/adapter.ts

```diff
@@ -166,8 +183,15 @@ export async function adapter(params: {
 }
 
 export function blockUnallowedResponse(
+  request: PreparedRequest['request'],
   promise: Promise<FetchEventResult>
 ): Promise<FetchEventResult> {
+  const pathname = request.nextUrl.pathname
+
+  if (pathname && (pathname.startsWith('/api/') || pathname === '/api')) {
```

Comment:
Can this support cases where the `/api` is hidden behind a rewrite like 
```
       {
          source: '/feedback-auth',
          destination: '/api/feedback-auth',
        },
```
?

Reply:
will add a test <picture data-single-emoji=":relieved-telegram:" title=":relieved-telegram:"><img class="emoji" width="20" height="auto" src="https://single-emoji.vercel.app/api/emoji/eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..z2vcFjSaddAGd65c.ncweoEeN_D70nUmHamDfqQkOI-b4Cp5P60eIZhwOSH5LMBaGcah6uCRgoc5ptGklraE6X363Rsu20ujRt0wvMB7bgZSO1sY26i81R_1WyPIoxYihjhhqju9A8cdjCyVqmg6m.su4DA3KMiukaPTL3ajAfpw" alt=":relieved-telegram:" align="absmiddle"></picture>

Reply:
I think that rewrites _do not_ work with middleware: we get the URL _before_ the rewrite <picture data-single-emoji=":thinkies:" title=":thinkies:"><img class="emoji" width="20" height="auto" src="https://single-emoji.vercel.app/api/emoji/eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..ZoJ60q0hVYl1dtYO.3AhDXxkwKHePIsrdW--wW_KKIFUtBydyi1GrhSAQCEEa-9Jf3YUZtzq2YINskm2yplkgDyFWJ23aVyrwhV7LL_J-qSMeDcLzVwrVTb4kgr2xDbuTT-sab1re.G3i78H61fppaLmoef3l28Q" alt=":thinkies:" align="absmiddle"></picture> <picture data-single-emoji=":hmmmmmmmmm:" title=":hmmmmmmmmm:"><img class="emoji" width="20" height="auto" src="https://single-emoji.vercel.app/api/emoji/eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..UNMLnTZ8L-B1QkoW._bC-pQP8oWX6b7ohUuN9SZvEuxOouFMdAf_6E-8y3mJ_RZcnRFBPhgjWwg9WbY_yLOjBNJx0hKmn-LqG9ItiBrkgLsXTxIQ6g6cuMYkZf-zEpiMeDcTxFUAUsxI.H9YyIpfATozIq8QAKeVRDQ" alt=":hmmmmmmmmm:" align="absmiddle"></picture>

Reply:
Yeah. What if we allow responses if the accept header does not start with `text/html` or is `*/*` (Next uses this for `_next/data`?

Reply:
so we will match against the `Accept` instead of the pathname, allowing to rewrite basically any route and not just API routes?

Reply:
Yep. Any route that isn't HTML or `next/data`. Also, we can likely identify `next/data` more affirmatively (based on the path)

Reply:
We already identify `next/data`: we have a `isDataReq` binding 😌 

so `!isDataReq && !acceptMatches("text/html") => response body is allowed`

Reply:
@cramforce following our Slack convo, can we resolve this? Do we want to stick to the pathname filtering?